pkg/resource/secret_reconciler.go

```diff
@@ -68,53 +68,51 @@ func NewSecretPropagatingReconciler(m manager.Manager) reconcile.Reconciler {
 		ctx, cancel := context.WithTimeout(context.Background(), secretReconcileTimeout)
 		defer cancel()
 
-		// The 'from' secret is also know as the 'propagating' secret.
-		from := &corev1.Secret{}
-		if err := client.Get(ctx, req.NamespacedName, from); err != nil {
-			// There's no propagation to be done if the secret we're propagating
-			// from does not exist. We assume we have a watch on that secret and
-			// will be queued if/when it is created. Otherwise we'll be requeued
-			// implicitly because we return an error.
-			return reconcile.Result{}, errors.Wrap(IgnoreNotFound(err), errGetSecret)
-		}
-
 		// The 'to' secret is also known as the 'propagated' secret. We guard
 		// against abusers of the propagation process by requiring that both
 		// secrets consent to propagation by specifying each other's UID. We
 		// cannot know the UID of a secret that doesn't exist, so the propagated
 		// secret must be created outside of the propagation process.
 		to := &corev1.Secret{}
-		n := types.NamespacedName{
-			Namespace: from.GetAnnotations()[AnnotationKeyPropagateToNamespace],
-			Name:      from.GetAnnotations()[AnnotationKeyPropagateToName],
-		}
-		if err := client.Get(ctx, n, to); err != nil {
+		if err := client.Get(ctx, req.NamespacedName, to); err != nil {
 			// There's no propagation to be done if the secret we propagate to
 			// does not exist. We assume we have a watch on that secret and will
 			// be queued if/when it is created. Otherwise we'll be requeued
 			// implicitly because we return an error.
 			return reconcile.Result{}, errors.Wrap(IgnoreNotFound(err), errGetSecret)
 		}
+		// The 'from' secret is also know as the 'propagating' secret.
+		from := &corev1.Secret{}
+		n := types.NamespacedName{
+			Namespace: to.GetAnnotations()[AnnotationKeyPropagateFromNamespace],
```

Comment:
I have removed the "invalid propagation format" errors in favor of the existing pattern of just checking whether we can get the from `Secret` using what is provided.